core/service/pool_test.go

```diff
@@ -30,12 +30,12 @@ type mockService struct {
 
 type mockPublisher struct {
 	publishedTopic string
-	publishedArgs  []interface{}
+	publishedArg   interface{}
```

Comment:
`Arg` was replaced to data everywhere, probably we should to it here too.
```suggestion
	publishedData   interface{}
```